packages/firebase_snippets_app/lib/snippets/auth.dart

```diff
@@ -0,0 +1,622 @@
+
+  void getStarted_idTokenChanges() async {
+    // [START get_started_id_token_changes]
+    FirebaseAuth.instance.idTokenChanges().listen((User? user) {
+      if (user == null) {
+        print('User is currently signed out!');
+      } else {
+        print('User is signed in!');
+      }
+    });
+    // [END get_started_id_token_changes]
+  }
+
+  void getStarted_userChanges() async {
+    // [START get_started_user_changes]
+    FirebaseAuth.instance.userChanges().listen((User? user) {
+      if (user == null) {
+        print('User is currently signed out!');
+      } else {
+        print('User is signed in!');
+      }
+    });
+    // [END get_started_user_changes]
+  }
+
+  void manageUsers_getProfile() async {
+    // [START manage_users_get_profile]
+    final user = FirebaseAuth.instance.currentUser;
+    if (user != null) {
+      // Name, email address, and profile photo URL
+      final name = user.displayName;
+      final email = user.email;
+      final photoUrl = user.photoURL;
+
+      // Check if user's email is verified
+      final emailVerified = user.emailVerified;
+
+      // The user's ID, unique to the Firebase project. Do NOT use this value to
+      // authenticate with your backend server, if you have one. Use
+      // User.getIdToken() instead.
+      final uid = user.uid;
+    }
+    // [END manage_users_get_profile]
+
+    // [START manage_users_get_provider_specific_profile]
+    if (user != null) {
+      for (final providerProfile in user.providerData) {
+        // ID of the provider (google.com, apple.cpm, etc.)
+        final provider = providerProfile.providerId;
+
+        // UID specific to the provider
+        final uid = providerProfile.uid;
+
+        // Name, email address, and profile photo URL
+        final name = providerProfile.displayName;
+        final emailAddress = providerProfile.email;
+        final profilePhoto = providerProfile.photoURL;
+      }
+    }
+    // [END manage_users_get_provider_specific_profile]
+
+    // [START manage_users_update_profile]
+    await user?.updateDisplayName("Jane Q. User");
+    await user?.updatePhotoURL("https://example.com/jane-q-user/profile.jpg");
+    // [END manage_users_update_profile]
+
+    // [START manage_users_set_email]
+    await user?.updateEmail("janeq@example.com");
+    // [END manage_users_set_email]
+
+    // [START manage_users_send_verification_email]
+    await user?.sendEmailVerification();
+    // [END manage_users_send_verification_email]
+
+    // [START manage_users_set_localization_then_send_email]
+    await FirebaseAuth.instance.setLanguageCode("fr");
+    await user?.sendEmailVerification();
+    // [END manage_users_set_localization_then_send_email]
+
+    const String newPassword = 'goodpassword';
+
+    // [START manage_users_set_new_password]
+    await user?.updatePassword(newPassword);
+    // [END manage_users_set_new_password]
+
+    // [START manage_users_send_reset_email]
+    await FirebaseAuth.instance
```

Comment:
The [corresponding web snippet](https://firebase.google.com/docs/auth/web/manage-users#send_a_password_reset_email) shows how to handle errors here, so let's how that for Flutter too. The closer to the other languages we keep this, the easier it'll be to make changes to the structure going forward.